### PR TITLE
Add a Contribution Guide

### LINKED ISSUE #359: Add CONTRIBUTING.md
As per the discussion in #248 we need to add a CONTRIBUTING.md to make this repository friendlier to newcomers. :)

### PR DESCRIPTION
**Pull request by capnkenny**
Closes #359 

Adds a new contribution guide as a standard for the NovelRT repository.

### CODE REVIEW

File: Contributing.md
Comment (FiniteReality) [inline Contributing.md:48]: I'd personally avoid emoji in the middle of sentences like this. To me it feels like something a five year old would write :sweat_smile: 

I think they'd be more useful to indicate important information, like notes which could use :warning: instead of `**Note:**` for example, but randomly placing them in the middle of a sentence is distracting at best IMO

File: Contributing.md
Comment (FiniteReality) [inline Contributing.md:158]: I'd probably disagree with this for the sole purpose of documentation - IMO creating a whole issue just to say "i think `teh` is a typo of `the` in this docs page" is a bit too much. Maybe we should clarify that we're specifically talking about code/functionality changes here?

File: Contributing.md
Comment (capnkenny) [reply Contributing.md:48]: Makes sense - tbh, I kept most of them from the Atom contribution guide I used as a basis, so we can remove them as they may distract at best, not make sense at worst

File: Contributing.md
Comment (capnkenny) [reply Contributing.md:158]: That works for me (calling out code/functionality changes w/o a PR)..

@RubyNova your take? You have final say, but I'm in @FiniteReality's boat here

Comment (RubyNova) [reply Contributing.md:158]: My only real concerns are if this causes conflicts in headers and creation of misaligned expectations. Logistically speaking it would be easier on us if someone who we knew was already editing a header file made minor documentation changes to make it easier on the merge conflicts. In addition to this, I'm just mildly concerned someone will be like "x got merged without an issue why not y" and then you're on a slippery slope.

How do we address these?

Comment (capnkenny) [reply Contributing.md:158]: TL;DR - Overall, I'd propose a matching Pull Request Template and Issue template (not in this PR but in a new one). Would address the issues, and would only require communication in Discord/GH prior to a PR being accepted.


Long version - Well, this depends on how we want to handle this holistically imo.

To be honest, after thinking it further, typos and such are still issues, therefore unless someone trusted is working on it, the person pushing for a fix should _probably_ just open the matching issue for it. On the contrary this is _going_ to inflate our issue count if we ever run into an issue where we're finding lots of typos, etc, so we'd run into the exact scenario that @FiniteReality mentioned. 
But at least if someone trusted is working on it, there's typically more communication in at least Discord, therefore there's less chance of collision / merge conflicts @RubyNova.

As far as this goes:
> "x got merged without an issue why not y"
We'd really just need to be diligent on ensuring contributors know what was being affected (documentation only) and _whom_ was affecting it (if going the route of letting documentation-style quick fixes go without a matching issue). 

With that said, we could introduce templates for PRs and Issues, and only require that code/functionality changes have a matching issue _as long as the template is followed and communication has occurred somewhere publicly visible_ with [insert someone with merge rights here]. This gives us an easier way to handle - anyone not following the template can be closed until they're following the format (once they're introduced - and we can have GH help notify contributors of this).

**Comment by capnkenny** [comment]
Hey @RubyNova - did we ever finalize what we were doing with the above? I know there was some back-and-forth however I don't remember where we landed.

**Comment by RubyNova** [comment]
I don't remember either. :D

File: Contributing.md
Comment (capnkenny) [reply Contributing.md:158]: Opened #426 to track this work - won't resolve until I reword some of this as per Monica's original suggestions.

Comment (capnkenny) [reply Contributing.md]: Resolved (hopefully) in latest commit

File: Contributing.md
Comment (RubyNova) [inline Contributing.md:35]:
```suggestion
* [Github Discussions](https://github.com/novelrt/NovelRT/discussions)
```

Maybe...? Unsure. It just read weird to me.

File: Contributing.md
Comment (RubyNova) [inline Contributing.md:36]:
```suggestion
* [NovelRT Discord Server](https://discord.novelrt.dev/)
```

File: Contributing.md
Comment (RubyNova) [inline Contributing.md:86]: We do not inherently support VMs - people using them are on their own.

File: Contributing.md
Comment (capnkenny) [reply Contributing.md]: Changed in latest

File: Contributing.md
Comment (RubyNova) [inline Contributing.md:97]: We should probably add that even if it doesn't explicitly fall under this, we reserve the right to do it anyway if we feel you're in violation.

File: Contributing.md
Comment (capnkenny) [reply Contributing.md]: Fixed - forgot we had this lol

File: Contributing.md
Comment (capnkenny) [reply Contributing.md]: Removed

File: Contributing.md
Comment (RubyNova) [inline Contributing.md:153]: We should probably add that the codebase is UK English and ideally spellings should try to be caught before we review it.

**Review by RubyNova — CHANGES_REQUESTED** [review]

File: Contributing.md
Comment (capnkenny) [reply Contributing.md:97]: `The NovelRT team also reserves the right to enforce disiplinary action for any reason that may violate our Code of Conduct at any time.`

Sounds good?

Comment (RubyNova) [reply Contributing.md:97]: Sounds good. :)

File: Contributing.md
Comment (capnkenny) [reply Contributing.md:153]: You caught a mistake with the numbering here too lol

**Review by RubyNova — APPROVED** [review]
LGTM

File: Contributing.md
Comment (FiniteReality) [reply Contributing.md]: FWIW I'm super down for supporting VM users as I myself run Windows in a VM to both develop and test NovelRT and Fabulist. But if you don't want to explicitly indicate support, I'm okay for providing that support independently.

**Pull request merged**
